### ISSUE TITLE
[Tama] [Apollo] Serious camera video frame rate issues

### ISSUE THREAD

**Issue by AntonioBiscuit** [sonyxperiadev/bug_tracker#615]
**Platform:** Tama 
**Device:** Apollo
**Kernel version:** 4.14
**Android version:** PixelExperience_h8324-10.0-20200714-1039-OFFICIAL
**Software binaries version:** V9C

**Previously working on**
Stock Android

**Description**
Camera preview refresh rate seems to be capped to around 15-20FPS.
Regardless of what definition or frame rate is set, recording video will result to a very choppy video file of around 15-20FPS.

**When possible to change, still had video and preview frame rate same issues with**
Different camera apps: Open Camera, Hedge Cam, FreeDCam, ROM's default Camera
Default Camera API and Camera 2 API
Lots or little amounts light
Auto/Manual Focus
Auto/Fixed White Balance
Auto/Fixed Shutter Speed
No Noise Reduction or any post-processing
HEVC and H264 codecs
24FPS, 30FPS frame rates
1080p/2160p

**How to reproduce**
Use the phone's camera and/or record a video to see preview and/or video frame rate issues.

**Camera Settings related to the image**
Recorded using Open Camera
Camera 2 API
Frame Rate: 30FPS
Bitrate: 100Mbit/s
Codec: HEVC
Definition: 2160p
Shutter: 1/60s
White balance: Daylight
Focus: Manual
Noise Reduction: None

![image](https://user-images.githubusercontent.com/64173083/87560064-59714180-c6bb-11ea-8a7e-44aa559973be.png)

**Comment by leeoniya** [comment]
maybe related to #603?

**Comment by AntonioBiscuit** [comment]
Hi, does anyone have news or clues about that issue ?

**Comment by AntonioBiscuit** [comment]
It just crossed my mind right now but I'm starting to think about a scheduler issue.
Couldn't the FPS issue come from restrained frequencies somewhere and also be related to #603 ?
Couldn't loading lagspikes when loading content from Twitter, YouTube or anything else it could be mitigated with a scheduler tuning ?

**Comment by sraase** [comment]
Does this appear at lower resolutions (1280x720, 640x480) as well?

**Comment by sraase** [comment]
This is a frame drop issue (obvious from the kernel log). Playing with the camera clocks did not help, but running the sensor at lower resolution fixes it completely. However, CamX doesn't like to do that.

As a workaround, you can use HFR mode (120 fps) instead, which is limited to 1280x720. In OpenCamera (Camera2), that is done by manually setting the frame rate to "60 [High Speed]" or "120 [High Speed]".

**Comment by sraase** [comment]
OpenCamera requests a 3840x2160 stream in almost all modes, forcing the sensor into high-resolution (17 MP) mode, causing bad performance. This cannot be fixed.

If the sensor is run in 1920x1080/60fps mode instead, framedrops still occur. As a workaround, it can be avoided by setting the following property. **Note that doing this may cause increased battery consumption and additional heat generation in all camera modes** (default value is zero):

> adb shell setprop persist.vendor.camera.ife.externalBandwidthMBytes 0xffffffff

In high-framerate mode, no framedrops occur. See previous post.

**Comment by faenil** [comment]
@sraase I still have this issue on Tama AOSP11 r46 with OEMv7 binaries.

I have decreased the back camera video resolution to 480p and it's still unusable.

This is using the default camera shipped with AOSP11

Am I doing anything wrong? Can we reopen this issue if this is confirmed to still be a problem?

Excerpt from logcat
```
02-27 08:31:27.532   702  3035 E CamX    : [ERROR][IQMod  ] camxipeica.cpp:1471 UpdatePerspectiveParamsToContext() Invalid warp data for path 0
02-27 08:31:27.534     0     0 I CAM_INFO: CAM-CRM: __cam_req_mgr_traverse: 261 Skip Frame: req: 9 not ready pd: 2 open_req count: 5
02-27 08:31:27.538   715  4630 I sensors-hal: handle_sns_std_sensor_event:45, ambient_light_event: filter out status = 0, lux = 36, raw_adc=0
02-27 08:31:27.540   702  3040 I CHISENSORSONY: IMX400: set gain 3.46 (analog 3.46 digital 1.00 isp 1.00 lineCount 11602)
02-27 08:31:27.541   702  3038 E CamX    : [ERROR][FD     ] camxfdmanagernode.cpp:3598 PopulateGravityData() NCS Gravity handle is NULL
02-27 08:31:27.572   702  3036 I CHISENSORSONY: IMX400: set gain 3.46 (analog 3.46 digital 1.00 isp 1.00 lineCount 11602)
02-27 08:31:27.581   832  2292 E Utils   : did not find width and/or height
02-27 08:31:27.584   832  2294 E Utils   : did not find width and/or height
02-27 08:31:27.589   702  3038 E CamX    : [ERROR][FD     ] camxfdmanagernode.cpp:3598 PopulateGravityData() NCS Gravity handle is NULL
02-27 08:31:27.590   702  3037 E CamX    : [ERROR][IQMod  ] camxipeica.cpp:1471 UpdatePerspectiveParamsToContext() Invalid warp data for path 0
02-27 08:31:27.590   702   702 I CHIUSECASE: [REQMAP ] chxadvancedcamerausecase.cpp:3286 ExecuteCaptureRequest() frame: 159  <==>  (chiFrameNum) chiOverrideFrameNum: 13
02-27 08:31:27.593   702  3038 I CamX    : [REQMAP][CORE   ] camxsession.cpp:2744 ProcessRequest() chiFrameNum: 13  <==>  requestId: 14  <==>  sequenceId: 13  <==> CSLSyncId: 14  -- JPEGEncodePreview_0
02-27 08:31:27.600  4039  4039 V CAM_FilmstripView: suggesting item bounds: 1080x2112
02-27 08:31:27.600  4039  4039 V CAM_FilmstripView: getRecycledView, recycled=false
02-27 08:31:27.604  4039  4039 V CAM_FilmstripView: recycleView
02-27 08:31:27.605  4039  4039 I CAM_FilmstripView: [ViewItem:0] mData.renderThumbnail()
02-27 08:31:27.606  4039  4039 D CAM_FilmstripView: onFilmstripItemInserted()
02-27 08:31:27.611   702  4723 I CHIUSECASE: [REQMAP ] chxadvancedcamerausecase.cpp:3286 ExecuteCaptureRequest() frame: 160  <==>  (chiFrameNum) chiOverrideFrameNum: 14
02-27 08:31:27.614   702  3040 I CamX    : [REQMAP][CORE   ] camxsession.cpp:2744 ProcessRequest() chiFrameNum: 14  <==>  requestId: 15  <==>  sequenceId: 14  <==> CSLSyncId: 15  -- JPEGEncodePreview_0
02-27 08:31:27.640  2489  2513 W MediaProvider: Using FUSE for /storage/emulated/0/DCIM/Camera/VID_20220227_083120.mp4
02-27 08:31:27.651  2489  2510 W MediaProvider: Using FUSE for /storage/emulated/0/DCIM/Camera/VID_20220227_083120.mp4
02-27 08:31:27.657   702  3040 I CHISENSORSONY: IMX400: set gain 2.90 (analog 2.89 digital 1.00 isp 1.00 lineCount 11602)
02-27 08:31:27.664   702  3036 E CamX    : [ERROR][IQMod  ] camxipeica.cpp:1471 UpdatePerspectiveParamsToContext() Invalid warp data for path 0
02-27 08:31:27.675  4039  4331 D skia    : --- Failed to create image decoder with message 'unimplemented'
02-27 08:31:27.676  4039  4331 D skia    : --- Failed to create image decoder with message 'unimplemented'
02-27 08:31:27.680   702  3038 E CamX    : [ERROR][FD     ] camxfdmanagernode.cpp:3598 PopulateGravityData() NCS Gravity handle is NULL
02-27 08:31:27.680   702  4723 I CHIUSECASE: [REQMAP ] chxadvancedcamerausecase.cpp:3286 ExecuteCaptureRequest() frame: 161  <==>  (chiFrameNum) chiOverrideFrameNum: 15
02-27 08:31:27.681   832  2292 E Utils   : did not find width and/or height
02-27 08:31:27.700     0     0 I CAM_INFO: CAM-CRM: __cam_req_mgr_traverse: 261 Skip Frame: req: 12 not ready pd: 2 open_req count: 5
02-27 08:31:27.711     0     0 I msm_vidc: info: Opening video instance: 0000000000000000, 1
02-27 08:31:27.784     0     0 I CAM_INFO: CAM-CRM: __cam_req_mgr_traverse: 261 Skip Frame: req: 12 not ready pd: 1 open_req count: 4
02-27 08:31:27.825     0     0 I CAM_INFO: CAM-CRM: __cam_req_mgr_traverse: 261 Skip Frame: req: 12 not ready pd: 1 open_req count: 5
02-27 08:31:27.685   702  3039 I CamX    : [REQMAP][CORE   ] camxsession.cpp:2744 ProcessRequest() chiFrameNum: 15  <==>  requestId: 16  <==>  sequenceId: 15  <==> CSLSyncId: 16  -- JPEGEncodePreview_0
02-27 08:31:27.686   832  2292 E Utils   : did not find width and/or height
02-27 08:31:27.688   834  3253 E MetadataRetrieverClient: failed to extract an album art
02-27 08:31:27.688  4039  4774 E MediaMetadataRetrieverJNI: getEmbeddedPicture: Call to getEmbeddedPicture failed.
02-27 08:31:27.689   832  2294 E Utils   : did not find width and/or height
02-27 08:31:27.867     0     0 I CAM_INFO: CAM-CRM: __cam_req_mgr_traverse: 261 Skip Frame: req: 12 not ready pd: 1 open_req count: 7
02-27 08:31:27.695   834  4777 I OMXClient: IOmx service obtained
02-27 08:31:27.696   839  4730 I OMXMaster: makeComponentInstance(OMX.qcom.video.decoder.avc) in android.hardwar process
02-27 08:31:27.700   832  2294 E Utils   : did not find width and/or height
02-27 08:31:27.706   839  4730 D PlatformConfig: getInstance: Enter
02-27 08:31:27.706   839  4730 D PlatformConfig: getInt32: Returning default
02-27 08:31:27.707   839  4730 D PlatformConfig: getInstance: Enter
02-27 08:31:27.707   839  4730 D PlatformConfig: getInt32: Returning default
02-27 08:31:27.707   839  4730 D PlatformConfig: getInstance: Enter
02-27 08:31:27.708   839  4730 D PlatformConfig: getInt32: Returning default
02-27 08:31:27.708   839  4730 D PlatformConfig: getInstance: Enter
02-27 08:31:27.709   839  4730 D PlatformConfig: getInt32: Returning default
02-27 08:31:27.709   839  4730 D PlatformConfig: getInstance: Enter
02-27 08:31:27.709   839  4730 D PlatformConfig: getInt32: Returning default
02-27 08:31:27.710   839  4730 D PlatformConfig: getInstance: Enter
02-27 08:31:27.710   839  4730 D PlatformConfig: getInt32: Returning default
02-27 08:31:27.710   702  3035 E CamX    : [ERROR][IQMod  ] camxipeica.cpp:1471 UpdatePerspectiveParamsToContext() Invalid warp data for path 0
02-27 08:31:27.711   839  4730 I OMX-VDEC-1080P: component_init: OMX.qcom.video.decoder.avc : fd=8
02-27 08:31:27.711   839  4730 E OMX-VDEC-1080P: Unsupported output color format for c2d (2141391876)
02-27 08:31:27.712   839  4730 E OMX-VDEC-1080P: Setting color format failed
02-27 08:31:27.718   702  3037 I CHISENSORSONY: IMX400: set gain 2.46 (analog 2.46 digital 1.00 isp 1.00 lineCount 11602)
02-27 08:31:27.720   702  3040 E CamX    : [ERROR][FD     ] camxfdmanagernode.cpp:3598 PopulateGravityData() NCS Gravity handle is NULL
02-27 08:31:27.726   839  4730 D PlatformConfig: getInstance: Enter
02-27 08:31:27.726   839  4730 D PlatformConfig: getInt32: Returning default
02-27 08:31:27.727   839  4730 D PlatformConfig: getInstance: Enter
02-27 08:31:27.727   839  4730 D PlatformConfig: getInt32: Returning default
02-27 08:31:27.728   839  4730 I OMX-VDEC-1080P: omx_vdec::component_init() success : fd=8
02-27 08:31:27.731   839  1871 E OMX-VDEC-1080P: Extension: OMX.google.android.index.storeANWBufferInMetadata not implemented
02-27 08:31:27.731   839  1871 E OMX-VDEC-1080P: Extension: OMX.google.android.index.configureVideoTunnelMode not implemented
02-27 08:31:27.732   839  1871 E OMX-VDEC-1080P: Extension: OMX.google.android.index.useAndroidNativeBuffer is supported
02-27 08:31:27.732   839  2136 E OMXNodeInstance: getParameter(0xed0d0024:qcom.decoder.avc, ??(0x7f000044)) ERROR: UnsupportedSetting(0x80001019)
02-27 08:31:27.733   839   949 E OMXNodeInstance: getParameter(0xed0d0024:qcom.decoder.avc, ??(0x7f000044)) ERROR: UnsupportedSetting(0x80001019)
02-27 08:31:27.734   834  4777 D ACodec  : Setting [OMX.qcom.video.decoder.avc] input port buffer number: 1
02-27 08:31:27.734   839  2136 E OMX-VDEC-1080P: ERROR: IP Requirements(#4: 8847360) Requested(#1: 8847360)
02-27 08:31:27.735   839  2136 E OMX-VDEC-1080P: set_parameter: Error: 0x80001005, setting param 0x2000001
02-27 08:31:27.735   839  2136 E OMXNodeInstance: setParameter(0xed0d0024:qcom.decoder.avc, ParamPortDefinition(0x2000001)) ERROR: BadParameter(0x80001005)
02-27 08:31:27.735   834  4777 W ACodec  : Fail to set [OMX.qcom.video.decoder.avc] input port buffer number: 1
02-27 08:31:27.736   834  4777 D ACodec  : Setting [OMX.qcom.video.decoder.avc] output port buffer number: 1
02-27 08:31:27.736   839   949 I OMX-VDEC-1080P: C2D: Set Resolution, Interlace(false) Conversion(0XB -> 0X4) src(320X240) dest(320X240)
02-27 08:31:27.739   839   949 I C2DColorConvert: getDummySurfaceDef: half stride for Cb Cr planes 
02-27 08:31:27.740   839  4730 E OMX-VDEC-1080P: OP Requirements: Client count(1) insufficient v/s requested(12)
02-27 08:31:27.740   839  4730 E OMX-VDEC-1080P: ERROR: OP Requirements(#12: 118784) Requested(#1: 118784)
02-27 08:31:27.740   839  4730 E OMX-VDEC-1080P: set_parameter: Error: 0x80001005, setting param 0x2000001
02-27 08:31:27.741   839  4730 E OMXNodeInstance: setParameter(0xed0d0024:qcom.decoder.avc, ParamPortDefinition(0x2000001)) ERROR: BadParameter(0x80001005)
02-27 08:31:27.741   834  4777 W ACodec  : Fail to set [OMX.qcom.video.decoder.avc] output port buffer number: 1
02-27 08:31:27.742   839  2136 I OMX-VDEC-1080P: C2D: Set Resolution, Interlace(false) Conversion(0XB -> 0X4) src(720X480) dest(720X480)
02-27 08:31:27.742   839  2136 I C2DColorConvert: getDummySurfaceDef: half stride for Cb Cr planes 
02-27 08:31:27.743   839  1871 I OMX-VDEC-1080P: C2D: Set Resolution, Interlace(false) Conversion(0XB -> 0X4) src(720X480) dest(720X480)
02-27 08:31:27.743   839  1871 I C2DColorConvert: getDummySurfaceDef: half stride for Cb Cr planes 
02-27 08:31:27.744   839  4283 I OMX-VDEC-1080P: C2D: Set Resolution, Interlace(false) Conversion(0XB -> 0X4) src(720X480) dest(720X480)
02-27 08:31:27.744   839  4283 I C2DColorConvert: getDummySurfaceDef: half stride for Cb Cr planes 
02-27 08:31:27.745   839   839 E OMX-VDEC-1080P: Extension: OMX.google.android.index.describeHDR10PlusInfo not implemented
02-27 08:31:27.746   839  4731 I OMX-VDEC-1080P: C2D: Set Resolution, Interlace(false) Conversion(0XB -> 0X4) src(720X480) dest(720X480)
02-27 08:31:27.747   839  4731 I C2DColorConvert: getDummySurfaceDef: half stride for Cb Cr planes 
02-27 08:31:27.747   839   839 E OMXNodeInstance: getParameter(0xed0d0024:qcom.decoder.avc, ??(0x7f000044)) ERROR: UnsupportedSetting(0x80001019)
02-27 08:31:27.760   702  3035 I CHISENSORSONY: IMX400: set gain 2.16 (analog 2.16 digital 1.00 isp 1.00 lineCount 11602)
02-27 08:31:27.764   839   949 I OMX-VDEC-1080P: C2D: Set Resolution, Interlace(false) Conversion(0XB -> 0X4) src(720X480) dest(720X480)
02-27 08:31:27.764   839   949 I C2DColorConvert: getDummySurfaceDef: half stride for Cb Cr planes 
```

**Comment by sraase** [comment]
> I have decreased the back camera video resolution to 480p and it's still unusable.

Check the configure_streams() call (you have to increase log verbosity). At least one stream is larger than 1920x1080, forcing the sensor into high resolution (low fps) mode. According to Wikipedia, Apollo's display resolution is 2160x1080, so this is probably the video preview stream. It cannot be configured in the AOSP Camera app (and most apps will behave the same).

> Am I doing anything wrong? Can we reopen this issue if this is confirmed to still be a problem?

It is not a "problem" because it works as designed. Try to find an app which allows you to configure all streams.

**Comment by faenil** [comment]
Great, thanks for the pointers @sraase, I will have a look!

>It is not a "problem" because it works as designed. Try to find an app which allows you to configure all streams.

For my understanding, how does it work in stock FW? 
It this a problem of the AOSP image requesting higher preview resolution than stock?
In other words, what is the difference with the stock Sony fw?
Is it feasible to get to a comparable level of smoothness as stock?

**Comment by sraase** [comment]
> For my understanding, how does it work in stock FW?

Stock uses a different HAL, blocks 3rd party apps from higher resolutions and does vendor magic with the camera app. (Which is why the Xperia camera app does not work anywhere else.)

> Is it feasible to get to a comparable level of smoothness as stock?

Get the sensor to 1080p60 and increase bandwidth and your preview will be smooth. No high resolution pictures in this mode, though.

> Does the AOSP Camera always use the full sensor resolution to display the preview (even on other non-Sony devices) or is this a problem with the userspace/HAL/kernel providing the wrong "preview resolution" hint?

It uses the *display* resolution for preview. There are no hints. Please look at the Google Camera2 API, the HAL layer works in a very similar fashion.

**Comment by faenil** [comment]
Thanks for the clarifications @sraase 

>No high resolution pictures in this mode, though.

Do we know how stock is able to have a smooth high resolution preview, with high res pics and smooth video?

What is the way forward for us (if any) to get high resolution pictures and smooth preview and video recording?
Is this something we can accomplish (by fixing some camera stack bits and/or using 3rd party apps) or is it only possible via proprietary information/binaries we will not get?

>It uses the display resolution for preview. There are no hints. Please look at the Google Camera2 API, the HAL layer works in a very similar fashion.

I had a brief look at the Camera2 API :+1: I will go through the docs when I have some time.

However, I am not sure I understand if there's anything we can do to fix this :sweat_smile: 

In other words, do we have a list of actionable points to get a camera experience similar to stock?
(I think users of AOSP users are okay with getting a lower pic quality, they just need a _usable_ camera)

**Comment by sraase** [comment]
> Do we know how stock is able to have a smooth high resolution preview, with high res pics and smooth video?

Preview is not high resolution, and the sensor is reconfigured for snapshots.

> What is the way forward for us (if any) to get high resolution pictures and smooth preview and video recording?

There is no way forward. Hence the closed ticket.

> In other words, do we have a list of actionable points to get a camera experience similar to stock?

There is no list of actionable points. If you want a stock camera experience, you can get it with stock.

> (I think users of AOSP users are okay with getting a lower pic quality, they just need a usable camera)

The Open Devices project does not promise a great camera experience. For reference, please look at https://developer.sony.com/develop/open-devices/get-started/supported-devices-and-functionality/current-platform-functionality-maintained/ (it has not been updated for a long time, but the camera feature list for the listed devices is approximately clear).

The camera works and delivers 1080p video. The definition of "usable" may vary, but fixing this issue is simply impossible using the current camera HAL and nobody is going to write a new HAL for this platform.

**Comment by faenil** [comment]
>Preview is not high resolution, and the sensor is reconfigured for snapshots.

Is that something that needs to happen in the HAL, or could we work that around with custom logic in a camera app?

Would it be a gargantuan effort to rewrite the Camera HAL?
Is there any possibility to investigate reusing the one from other devices with similar SoC that might have been released as OSS?
(maybe not all functionality will work, but that might be an acceptable compromise)

>The Open Devices project does not promise a great camera experience

I am not implying it is their responsibility to provide it.
On the contrary, I am volunteering my time to making that happen, if that is something that can be done with the available information.

However I sense the offer for help is not really welcome :sweat_smile:

**Comment by sraase** [comment]
> However I sense some resistance

It is more that I have very limited time... and that I don't think we should have this discussion publicly. Please contact me directly.

**Comment by faenil** [comment]
Awesome, will get in touch, thanks @sraase !

**Comment by MarijnS95** [comment]
> I am volunteering my time to making that happen

This sounds like something that should be fixed _properly_, _once_, on the mainline Linux kernel (and be effective and reusable for eternity).

Unfortunately V4L2 (the kernel API that is currently used for cameras) isn't well suited to the hardware layout used in our phones, similar to how this Android HAL isn't.

**Comment by faenil** [comment]
@sraase here's the output of configure_streams when opening AOSP Camera app

```
02-28 14:35:12.109   706  1569 E CamX    : [ERROR][CORE   ] camxnode.cpp:6246 FinalizeInitialization() Node::PreviewVideo_FDHw0 maxImageBuffers(12) from link properties should be atleast requiredMaxImageBuffers(14). So replacing maxImageBuffers with requiredMaxImageBuffers
02-28 14:35:12.112   706  1569 I CamX    : [CONFIG][STATS_AF] camxautofocusnode.cpp:300 FinalizeBufferProperties() PDAFType: 0, PDAFPortEnable: 1, SensorPDAFSupport 0, pPDCallback: 0x723ab2eca0, PDAF Enabled: 0
02-28 14:35:12.113   706  1569 I CamX    : [ INFO][STATS_AEC] qcaecalgorithminterface.cpp:593 AECAlgoSetNodeInterface() VendorTagBaseForAF: 2152988672
02-28 14:35:12.172   706  1569 I CamX    : [CONFIG][SENSOR ] camxsensornode.cpp:997 PrepareStreamOn() sensorConfigStatus:5 isEarlyPCREnabled:0
02-28 14:35:12.173   706  1569 I CamX    : [CONFIG][CORE   ] camxpipeline.cpp:733 Link() [PreviewVideo_0] Link for pipeline: 0x743d5afeb0 Success. Link = 0x743d5b1e38
02-28 14:35:12.173   706  1569 I CamX    : [CONFIG][CORE   ] camxpipeline.h:1800 SetPipelineStatus() PreviewVideo_0 status is now PipelineStatus::FINALIZED
02-28 14:35:12.174   706  1569 I CamX    : [CONFIG][CORE   ] camxsession.cpp:1233 Initialize() Session (0x72336d6040) Initialized
02-28 14:35:12.185   706  1569 I CamX    : [CONFIG][HAL    ] camxhal3.cpp:980 configure_streams()  FINAL stream[0] = 0x733d01d0f8 - info:
02-28 14:35:12.185   706  1569 I CamX    : [CONFIG][HAL    ] camxhal3.cpp:983 configure_streams()             format       : 35, HAL_PIXEL_FORMAT_YCbCr_420_888
02-28 14:35:12.186   706  1569 I CamX    : [CONFIG][HAL    ] camxhal3.cpp:985 configure_streams()             width        : 1920
02-28 14:35:12.186   706  1569 I CamX    : [CONFIG][HAL    ] camxhal3.cpp:987 configure_streams()             height       : 1440
02-28 14:35:12.186   706  1569 I CamX    : [CONFIG][HAL    ] camxhal3.cpp:990 configure_streams()             stream_type  : 00000000, CAMERA3_STREAM_OUTPUT
02-28 14:35:12.187   706  1569 I CamX    : [CONFIG][HAL    ] camxhal3.cpp:992 configure_streams()             usage        : 00020003
02-28 14:35:12.187   706  1569 I CamX    : [CONFIG][HAL    ] camxhal3.cpp:994 configure_streams()             max_buffers  : 8
02-28 14:35:12.187   706  1569 I CamX    : [CONFIG][HAL    ] camxhal3.cpp:997 configure_streams()             rotation     : 00000000, CAMERA3_STREAM_ROTATION_0
02-28 14:35:12.188   706  1569 I CamX    : [CONFIG][HAL    ] camxhal3.cpp:1000 configure_streams()             data_space   : 08c20000, HAL_DATASPACE_V0_JFIF
02-28 14:35:12.188   706  1569 I CamX    : [CONFIG][HAL    ] camxhal3.cpp:1002 configure_streams()             priv         : 0x737d02dfb0
02-28 14:35:12.188   706  1569 I CamX    : [CONFIG][HAL    ] camxhal3.cpp:1004 configure_streams()             reserved[0]         : 0x737d02dfb8
02-28 14:35:12.189   706  1569 I CamX    : [CONFIG][HAL    ] camxhal3.cpp:1006 configure_streams()             reserved[1]         : 0x0
02-28 14:35:12.189   706  1569 I CamX    : [CONFIG][HAL    ] camxhal3.cpp:1020 configure_streams()    pHalStream: 0x737d02dfb8 format : 0x23, overrideFormat : 0x23 consumer usage: 0, producer usage: 20003
02-28 14:35:12.189   706  1569 I CamX    : [CONFIG][HAL    ] camxhal3.cpp:980 configure_streams()  FINAL stream[1] = 0x733d01df68 - info:
02-28 14:35:12.190   706  1569 I CamX    : [CONFIG][HAL    ] camxhal3.cpp:983 configure_streams()             format       : 34, HAL_PIXEL_FORMAT_IMPLEMENTATION_DEFINED
02-28 14:35:12.190   706  1569 I CamX    : [CONFIG][HAL    ] camxhal3.cpp:985 configure_streams()             width        : 1440
02-28 14:35:12.190   706  1569 I CamX    : [CONFIG][HAL    ] camxhal3.cpp:987 configure_streams()             height       : 1080
02-28 14:35:12.190   706  1569 I CamX    : [CONFIG][HAL    ] camxhal3.cpp:990 configure_streams()             stream_type  : 00000000, CAMERA3_STREAM_OUTPUT
02-28 14:35:12.191   706  1569 I CamX    : [CONFIG][HAL    ] camxhal3.cpp:992 configure_streams()             usage        : 10020100
02-28 14:35:12.191   706  1569 I CamX    : [CONFIG][HAL    ] camxhal3.cpp:994 configure_streams()             max_buffers  : 8
02-28 14:35:12.191   706  1569 I CamX    : [CONFIG][HAL    ] camxhal3.cpp:997 configure_streams()             rotation     : 00000000, CAMERA3_STREAM_ROTATION_0
02-28 14:35:12.192   706  1569 I CamX    : [CONFIG][HAL    ] camxhal3.cpp:1000 configure_streams()             data_space   : 00000000, HAL_DATASPACE_UNKNOWN
02-28 14:35:12.192   706  1569 I CamX    : [CONFIG][HAL    ] camxhal3.cpp:1002 configure_streams()             priv         : 0x737d0304d0
02-28 14:35:12.192   706  1569 I CamX    : [CONFIG][HAL    ] camxhal3.cpp:1004 configure_streams()             reserved[0]         : 0x737d0304d8
02-28 14:35:12.193   706  1569 I CamX    : [CONFIG][HAL    ] camxhal3.cpp:1006 configure_streams()             reserved[1]         : 0x0
02-28 14:35:12.193   706  1569 I CamX    : [CONFIG][HAL    ] camxhal3.cpp:1020 configure_streams()    pHalStream: 0x737d0304d8 format : 0x22, overrideFormat : 0x7fa30c06 consumer usage: 0, producer usage: 10020100
02-28 14:35:12.194   706  1569 I CamX    : [CONFIG][HAL    ] camxhal3.cpp:1026 configure_streams() HalOp: End CONFIG, logicalCameraId: 0, cameraId: 0
02-28 14:35:12.199   706  1572 I CamX    : [CONFIG][CORE   ] camxpipeline.h:1800 SetPipelineStatus() PreviewVideo_0 status is now PipelineStatus::RESOURCES_ACQUIRED
02-28 14:35:12.200   706  1572 I CamX    : [CONFIG][CORE   ] camxpipeline.cpp:568 StreamOn() [PreviewVideo_0] StreamingOn for pipeline: 0x743d5afeb0
02-28 14:35:12.206   706  1565 W CamX    : [ WARN][FD     ] camxfdenginelibwrapper.cpp:2743 FDELibWrapperCameraNNSetup() Set DSP power TURBO failed! nnResult = 44
02-28 14:35:12.207   706  1565 W CamX    : [ WARN][FD     ] camxfdenginelibwrapper.cpp:684 FDELibWrapperCreateDLHandle() Failure in FDELibWrapperCameraNNSetup()
02-28 14:35:12.207   706  1565 W CamX    : [ WARN][FD     ] camxfdengine.cpp:242 FDEngineCreate() Create DL handle failure, use standard solution, engineType 3, result = 1
02-28 14:35:12.220   706  1572 I CamX    : [CONFIG][CORE   ] camxpipeline.h:1800 SetPipelineStatus() PreviewVideo_0 status is now PipelineStatus::STREAM_ON
02-28 14:35:12.236   706  1568 I CamX    : [REQMAP][CORE   ] camxsession.cpp:2744 ProcessRequest() chiFrameNum: 0  <==>  requestId: 1  <==>  sequenceId: 0  <==> CSLSyncId: 1  -- PreviewVideo_0
```

**Comment by sraase** [comment]
stream[0] = 1920x1440, stream[1] = 1440x1080.
Since 1920x1440 is larger than 1920x1080, the sensor is forced into high-res mode.
